tests/ui/map_identity.rs

```diff
@@ -0,0 +1,23 @@
+// run-rustfix
+#![warn(clippy::map_identity)]
+#![allow(clippy::needless_return)]
+
+fn main() {
+    let x: [u16; 3] = [1, 2, 3];
+    // should lint
+    let _: Vec<_> = x.iter().map(|x| { return x }).collect();
+    let _: Vec<_> = x.iter().map(identity).collect();
+    let _: Option<u8> = Some(3).map(|x| x);
```

Comment:
we should also have a test for something like `x.iter().map(|x| x).map(|y| y)` imo.

Reply:
good idea, added a test for that!